python/scripts/usj2usx.py

```diff
@@ -0,0 +1,180 @@
+                end_node.set('eid', prev_chapter_sid)    
+            prev_chapter_sid = json_node['sid']
+
+    if json_node['type'] in ['table:row', 'table:cell']:
+        json_node['type'] = json_node['type'].replace("table:", "")
+
+    if json_node['type'] == 'optbreak':
+        json_node['type'] = "para"
+    new_node = lxml.etree.SubElement(usx_head, json_node['type'])
+    if 'marker' in json_node:
+        new_node.set('style', json_node['marker'])
+    for key in json_node:
+        if key not in ['type', 'marker', 'content']:
+            new_node.set(key, json_node[key])
+    if 'content' in json_node:
+        for item in json_node['content']:
+            if isinstance(item, str):
+                if len(new_node) == 0:
+                    new_node.text = item
+                else:
+                    new_node[-1].tail = item
+            else:
+                convert_usj(item, new_node)
+
+def usj_to_xml(input_usj):
+    '''Build the XML tree'''
+    global prev_verse_sid
+    global prev_chapter_sid
+    prev_verse_sid = None
+    prev_chapter_sid = None
+    output_usx = lxml.etree.Element('usx')
+    output_usx.set('version', '3.0')
+    for item in input_usj['content']:
+        convert_usj(item, output_usx)
+
+    # adding end nodes for verse and chapter as those are not in USJ
+    if prev_verse_sid is not None:
+        end_node = lxml.etree.Element('verse')
+        end_node.set('eid', prev_verse_sid)
+        last_para = find_para_node_for_verse_eid(output_usx)
+        last_para.append(end_node)
+    if prev_chapter_sid is not None:
+        end_node = lxml.etree.SubElement(output_usx, 'chapter')
+        end_node.set('eid', prev_chapter_sid)    
+
+    return output_usx
+
+def test():
+    '''Test by comparing the generated USX against origin.xml in test suite'''
+    from lxml.doctestcompare import LXMLOutputChecker
+    checker = LXMLOutputChecker()
+    usj_paths = glob.glob('tests/*/*/origin.json') + glob.glob('tests/*/*/*/origin.json')
+    conversion_error = 0
+    comparison_error = 0
+    for path in usj_paths:
+        with open(path, 'r', encoding='utf-8') as usj_file:
+            input_usj = json.load(usj_file)
+            # validate(instance=input_usj, schema=USJ_SCHEMA)
+            try:
+                output_usx = usj_to_xml(input_usj)
+            except Exception as exce:
+                conversion_error += 1
+                print(f"\n\nError at {path}\n{exce}")
+                continue
+                # raise Exception(f"Error at conversion on {path}") from exce
+        with open(path.replace('.json','.xml'), 'r', encoding='utf-8') as origin_file:
+            origin_usx = lxml.etree.parse(origin_file).getroot()
+            all_vid_nodes = origin_usx.findall(".//*[@vid]")
+            all_closed_nodes = origin_usx.findall(".//*[@closed]")
+            all_status_nodes = origin_usx.findall(".//*[@status]")
+            for node in all_vid_nodes:
+                del node.attrib['vid']
```

Comment:
That's cheating! :) The USFMTC will debate this.

Reply:
:) 
Are you saying that about `vid`s alone or for `closed` and `status` as well? 
Should we keep `vid`s ?

Reply:
BTW, this hack is done in the testing code before comparing, since I did not include vid in USJ.